src/session.rs

```diff
@@ -149,14 +145,59 @@ pub fn start(config: &Config, initial_request: Option<String>) -> i32 {
     0
 }
 
-/// When server is not running it's better to cancel blocking request.
-/// Because server can take a long time to initialize or can fail to start.
-/// We assume that it's less annoying for user to just repeat command later
-/// than to wait, cancel, and repeat.
-fn cancel_blocking_request(fifo: String) {
-    debug!("Blocking request but LSP server is not running");
-    let command = "lsp-show-error 'language server is not running, cancelling blocking request'";
-    std::fs::write(fifo, command).expect("Failed to write command to fifo");
+/// Tries to send an error to the client about a request that failed to parse.
+fn handle_broken_editor_request(
+    to_editor: &Sender<EditorResponse>,
+    request: String,
+    session: &str,
+) {
+    // Try to parse enough of the broken toml to send the error to the editor.
+    let client_re = regex::Regex::new(r#"(?m)^client\W*=\W*"([^"\n]+)""#)
```

Comment:
even though it won't matter in practice, I think it's good style to compile this effectively-const regex only once using `lazy_static`